testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/OIDCClientRegistrationTest.java

```diff
@@ -284,6 +284,22 @@ public void createPublicClient() throws ClientRegistrationException {
         Assert.assertNull(kcClientRep.getSecret());
     }
 
+    @Test
+    public void shouldReturnSecretWhenCreateClientWithClientSecretJWT() throws ClientRegistrationException {
+        OIDCClientRepresentation clientRep = createRep();
+        clientRep.setGrantTypes(Collections.singletonList(OAuth2Constants.CLIENT_CREDENTIALS));
+        clientRep.setTokenEndpointAuthMethod(OIDCLoginProtocol.CLIENT_SECRET_JWT);
+
+        OIDCClientRepresentation response = reg.oidc().create(clientRep);
+        Assert.assertEquals("client_secret_jwt", response.getTokenEndpointAuthMethod());
+        Assert.assertNotNull(response.getClientSecret());
+        Assert.assertNotNull(response.getClientSecretExpiresAt());
+
+        ClientRepresentation kcClientRep = getKeycloakClient(response.getClientId());
+        Assert.assertFalse(kcClientRep.isPublicClient());
+        Assert.assertNotNull(kcClientRep.getSecret());
```

Comment:
Is it possible to extend this test a bit and add also "negative" test when secret is not displayed when it should not be displayed? I suggest to do one more client update and possibly switch the method of the client authentication to `OIDCLoginProtocol.TLS_CLIENT_AUTH` and then check that secret and secretExpiresAt are null after this update? This should work once you do the changes proposed above.

Reply:
Sure, it makes sense to me.
I will add an update to test it.
Thank you.